feature-libs/product/configurators/common/src/components/config-group-menu/config-group-menu.component.ts

```diff
@@ -1,84 +1,91 @@
-import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
+import { ChangeDetectionStrategy, Component } from '@angular/core';
 import {
   Configurator,
   ConfiguratorCommonsService,
   ConfiguratorGroupsService,
 } from '@spartacus/core';
 import { Observable, of } from 'rxjs';
 import { delay, map, switchMap, take } from 'rxjs/operators';
-import { ICON_TYPE } from '../../../../cms-components/misc/icon/index';
-import { HamburgerMenuService } from '../../../../layout/header/hamburger-menu/hamburger-menu.service';
-import { ConfigRouterExtractorService } from '../../generic/service/config-router-extractor.service';
+import {
+  ConfigRouterExtractorService,
+  ConfigurationRouter,
+  HamburgerMenuService,
+} from '@spartacus/storefront';
+import { ICON_TYPE } from '@spartacus/storefront';
 
 @Component({
   selector: 'cx-config-group-menu',
   templateUrl: './config-group-menu.component.html',
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class ConfigGroupMenuComponent implements OnInit {
-  configuration$: Observable<Configurator.Configuration>;
-  currentGroup$: Observable<Configurator.Group>;
-
-  displayedParentGroup$: Observable<Configurator.Group>;
-  displayedGroups$: Observable<Configurator.Group[]>;
+export class ConfigGroupMenuComponent {
+  routerData$: Observable<
+    ConfigurationRouter.Data
+  > = this.configRouterExtractorService.extractRouterData();
+
+  configuration$: Observable<
+    Configurator.Configuration
+  > = this.routerData$.pipe(
+    switchMap((routerData) =>
+      this.configCommonsService.getConfiguration(routerData.owner)
+    )
+  );
+
+  currentGroup$: Observable<Configurator.Group> = this.routerData$.pipe(
+    switchMap((routerData) =>
+      this.configuratorGroupsService.getCurrentGroup(routerData.owner)
+    )
+  );
+
+  displayedParentGroup$: Observable<
+    Configurator.Group
+  > = this.configuration$.pipe(
+    switchMap((configuration) =>
+      this.configuratorGroupsService.getMenuParentGroup(configuration.owner)
+    ),
+    switchMap((parentGroup) => this.getCondensedParentGroup(parentGroup))
+  );
+
+  displayedGroups$: Observable<
+    Configurator.Group[]
+  > = this.displayedParentGroup$.pipe(
+    switchMap((parentGroup) => {
+      return this.configuration$.pipe(
+        map((configuration) => {
+          if (parentGroup) {
+            return this.condenseGroups(parentGroup.subGroups);
+          } else {
+            return this.condenseGroups(configuration.groups);
+          }
+        }),
+        delay(0)
+      );
+    })
+  );
 
   iconTypes = ICON_TYPE;
+  GROUPSTATUS = Configurator.GroupStatus;
```

Comment:
Let's follow the convention, and use camil case for variable names.

Reply:
Done